builtin/providers/openstack/resource_openstack_lb_listener_v2.go

```diff
@@ -139,7 +139,24 @@ func resourceListenerV2Create(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	log.Printf("[DEBUG] Create Options: %#v", createOpts)
-	listener, err := listeners.Create(networkingClient, createOpts).Extract()
+	var listener *listeners.LoadBalancerID
+	err = resource.Retry(10*time.Minute, func() *resource.RetryError {
+		var err error
+		log.Printf("[DEBUG] Attempting to create listener")
+		listener, err := listeners.Create(networkingClient, createOpts).Extract()
+		if err != nil {
+			if errCode, ok := err.(gophercloud.ErrUnexpectedResponseCode); ok {
+				if errCode.Actual == 409 || errCode.Actual == 500 {
+					log.Printf("[DEBUG] OpenStack listener is still creating.")
```

Comment:
Message is incorrect.  It's not really the listener that is still creating.  It's the load balancer that is either still creating or updating from a previous operation.